### DISCUSSION TITLE
Last 12 months generation & consumption

### DISCUSSION THREAD

**Discussion by grest** [PiotrMachowski/Home-Assistant-custom-components-Tauron-AMIplus#63] [Ideas]
For old generation of prosumers, for which 0.8 of exported generation is to be imported for rolling period of 12 months, it would be good to know what is the generation and consumption for last 12 months. It could be extracted from https://elicznik.tauron-dystrybucja.pl/odczyty page.

**Comment by grest** [comment]
zadzwoniłem na infolinie, to musiałoby chyba wyglądać trochę inaczej - chyba tak jak jest zaproponowane w innej dyskusji o ostatnich 12 miesiącach

**Reply by essenemari** [reply]
Też mnie intryguje jak to rozliczać i ładnie podawać w HA. Jestem prosumentem na starych zasadach.
Myślę, że opieranie się na danych generacji i poboru z ostatnich 12M jest błędne. 
Poprawcie mnie proszę jeśli się mylę. Encje typu "last_12_months_energy_*" prawdopodobnie zwracają wartość typu: "Dziś - 12M".  

Mam PV nieco ponad rok, niedawno dostałem fakturę rozliczeniową. Obejmuje ona okres 09/12/2021 do 30/11/2022. Jest także informacja "że do wykorzystania z bieżącego okresu rozliczeniowego pozostało 308kWh przez 12 miesięcy".

Rozumiem to tak, że 308kWh muszę wykorzystać do 30/11/2023. Jeśli na ów dzień będę miał nadwyżkę, to te 308kWh przepadnie. (?)

Rzeczywisty stan kWh do odebrania na dzień dzisiejszy to w moim przypadku będzie: 
((oddaneTenRok*0,8)-pobraneTenRok)+stanMagazynuFakturaRozl+bilansOkrRozlDoTegoRoku. 

Interesujące jest to jak naliczany jest stan magazynowy. W moim przypadku te 308kWh. Podejrzewam, że to ta wartość naliczana jest miesiąc do miesiąca i również odliczana miesiąc do miesiąca następnego okr. rozliczeniowego, w moim przypadku następnego roku. Obliczanie tego miesiąc do miesiąca jest możliwe dzięki integracji jest możliwe (wyciąganie danych z bazy).

Jak Wy rozumiecie rozl. Tauron na starych zasadach lub jak to trzeba liczyć żeby było dobrze?

PS
@PiotrMachowski Twoja integracja jest WYŚMIENITA! Dzięki!

**Comment by mattPiratt** [comment]
Witam.
widzę że najnowsza wersja dodatku pana Piotra posiada encje z 12months... oraz pozwala wybrać datę startu okresu rozliczeniowago podczas konfiguracji dodatku w HA. Natomiast przez prak przykładów jak z tego skorzystac, nie potrafię dojść do ładu (liczby jakie otrzynuję są nie spójne z eLicznik stroną).
Czy można by prosić panie @PiotrMachowski o dokumentację do tego pola z datą, i jak powinno ono wpływać na encje?

Pozdrawiam serdecznie i kawka wysłana za świetną pracę !

**Reply by PiotrMachowski** [reply]
@mattPiratt encje "configurable" przedstawiają zsumowane wartości dla zakresu od wybranej daty do teraz. Encje "12 months" przedstawiają sumę z zakresu ostatnich 12 miesięcy (np. dzisiaj to jest zakres 24.01.2022-24.01.2023)

**Reply by essenemari** [reply]
@mattPiratt to co ostatnio dodał Piotr jest bardzo przydatne tj. wybranie startu okresu rozliczeniowego. Wcześniej było tak, że aby wyliczyć sobie bilans energii na dany dzień okr. rozl., pasowało np. użyć encji 'pobrano/oddano obecny rok' (bo takie były zaciągane z eLicznik) i uwzględnić lukę wynikającą z okresu rozliczeniowego prosumenta (każdy ma różny) a obecnym rokiem kalendarzowym. Teraz ta wartość jest podliczana automatycznie co jest bardzo wygodne i upraszcza obliczenia własne.

**Reply by mattPiratt** [reply]
@essenemari Witam i dziękuję za kontakt. Właśnie wczoraj wieczorem udało mi się to z konfigurować dokładnie tak jak koledzy opisują. I tycznie jest to duża ulga i ułatwienie.

Dodam że dawniej posiłowałem się czyś co się nazywa utility_meter, i w HA można ustawić tego stan początkowy, a reszta liczy się wg ustalonych cykli.
```
utility_meter:
  qm_meter_daily_energy_produced:
    source: sensor.qm_tpl_pv_energy_lifetime
    cycle: daily
  qm_meter_monthly_energy_produced:
    source: sensor.qm_tpl_pv_energy_lifetime
    cycle: monthly
  qm_meter_yearly_energy_produced:
    source: sensor.qm_tpl_pv_energy_lifetime
    cycle: yearly
```

To czego brakuje mi aktualnie i nie mam na to pomysłu to jak porównać produkcję i pobór rok do roku (czyli np styczeń 2023 do styczeń 2022). Tego typu wykres słupkowy byłby interesujący.

Natomiast podstawowa potrzeba pokazania bilansu rozliczenia aktualnego okresu rozliczeniowego jest i działa świetnie. Brawo dla twórcy @PiotrMachowski

**Reply by PiotrMachowski** [reply]
> To czego brakuje mi aktualnie i nie mam na to pomysłu to jak porównać produkcję i pobór rok do roku (czyli np styczeń 2023 do styczeń 2022). Tego typu wykres słupkowy byłby interesujący.

Tego niestety jeszcze nie ma w HA, ale można to zrobić na stronie eLicznika

**Reply by essenemari** [reply]
> To czego brakuje mi aktualnie i nie mam na to pomysłu to jak porównać produkcję i pobór rok do roku (czyli np styczeń 2023 do styczeń 2022). Tego typu wykres słupkowy byłby interesujący.

A czy sekcja w HA "energia" nie spełniła by Twoich oczekiwań? Ja u siebie widzę coś takiego i wygląda na to czego szukasz. Te dwa pierwsze słupki obok siebie to porównanie stycznia 2022 i 2023.
![image](https://user-images.githubusercontent.com/25403680/215437188-8b69bc15-e52a-4fef-a160-1061aeb8f341.png)

**Reply by essenemari** [reply]
Coś jednak nie działa to idealnie, bo po analizie te dwa pierwsze słupki to 01.2023 i 02.2022. Podejrzewam, że u mnie coś nie tak. Może mam mam za mało danych albo dlatego, że styczeń jeszcze trwa. Co do zasady to chyba powinno tak pokazywać jak to czego szukasz.

**Comment by dariuszszwajcok** [comment]
Witam. Podpowiedzcie mi jak ustawić okres rozliczeniowy od 1.04 do rocznego okresu rozliczeniowego bo już zgłupiałem w konfigurowalnym oddaniu, pobraniu i rozliczeniu... bede wdzieczny. Wcześniej miałem to zrobione poprzez [crona](cron: "0 0 1 4 *") teraz widze że jest możliwość konfiguracji w prost sposob który mnie rozłożył :)

**Reply by PiotrMachowski** [reply]
Po dodaniu integracji klikasz przycisk "konfiguruj", a następnie włączasz "konfigurowalne sensory" i podajesz wybraną datę. To samo można też zrobić podczas pierwszej konfiguracji

**Reply by dariuszszwajcok** [reply]
dzieki! nie widziałem w konfiguracji bo robiłem tylko aktualizacje integracji przez hacs ;)   śmiga !

**Reply by dariuszszwajcok** [reply]
Zawuważyłem dziś u siebie dziwne zachowanie tj:
oddanie 3032kWh
pobrano 2582kWh
bilans    -450

Wynika z tego że nie wzieło pod uwage wspołczynnika 0,8
Wg mojego przelicznika wychodzi mi:
oddano 3032               x 0,8 = 2425kWh
pobranno 2582
bilans -157

Ale jest też jedna dziwna rzecz mianowicie z powodu problemów huawei z oprogramowaniem optymalizatorow produkcja wynosi  0 kWh, zatem w ciągu ostatniej doby miałem:
oddanie 0 kWh
pobranie 14,6 kWh
bilans po przeliczeniu z wczorajszego dnia zmienił się z  UWAGA  z  -465,3  na  -450,9 kWh.  Wynika z tego że zamiast pobrać 14,6 to je oddałem...   dni poprzednie tez wygladały podobnie bo dwa dni temu mialem bilans -478 kWh nie oddając nic do sieci tylko pobierając.

state_class: total
tariff: G11
sum_consumption: 2131.325
sum_generation: -2582.298                   <------------------------  czy tutaj nie powinna być wartość dodatnia?
data_range: 2022-04-01 - 2023-01-27
cała doba_consumption: 2131.325
cała doba_generation: -2582.298
unit_of_measurement: kWh
device_class: energy
icon: mdi:counter
friendly_name: Tauron AMIplus Axzyxzyxzyxz  Configurable balance



Czy tylko ja tak mam ze nie sprowadza mi wartości oddanej do 80%?
Czy też miał ktoś problem z naliczaniem bilansu jak u mnie?
z góry dzieki za pomoc

**Reply by mattPiratt** [reply]
@dariuszszwajcok Witaj. Z tego co rozumiem to AMIplus moduł podaje ci tylko odczyty z serwisu eLicznik: oddano, pobrano, bilans.
Wiec pytanie co uwzglednia bilans nalezy kierowac do taurona. Wg moich porównań, to bilans nie uwzględnia współczynnika 0,8 ani 0,7, tylko podaje czysy bilans. Rachunek pomniejszający  *0,8 należy samemu zrobic.

**Reply by PiotrMachowski** [reply]
@mattPiratt Bilans nie jest obliczany jako zużycie - 0.8 * produkcja, tylko wg bilansu godzinowego (więcej opisu tu: https://github.com/PiotrMachowski/Home-Assistant-custom-components-Tauron-AMIplus/issues/57)

**Comment by cyzyk** [comment]
Mam problem nie mogę zmienić daty okresu rozliczeniowego
![tauron](https://github.com/user-attachments/assets/4f99b808-1748-47cf-970f-1b3685d1d72a)

Ciągle zostaje data 01.04.2024, a chce zmienić na 28.02.2025, może gdzieś w konfiguracji mogę zmienić ręcznie ?

**Reply by PiotrMachowski** [reply]
Utwórz issue, załącz logi
